src/nvim/terminal.c

```diff
@@ -793,6 +794,57 @@ static VTermKey convert_key(int key, VTermModifier *statep)
     case K_KMULTIPLY: return VTERM_KEY_KP_MULT;
     case K_KDIVIDE:   return VTERM_KEY_KP_DIVIDE;
 
+    case K_S_F1:      *statep |= VTERM_MOD_SHIFT;
```

Comment:
perhaps all of these `K_S_*` variants should be grouped in a separate switch before this one.

```c
switch(key) {
    case K_S_F2:
    case K_S_F3:
    case K_S_F4:
    ...
      *statep |= VTERM_MOD_SHIFT;
}